pandas/core/ops/__init__.py

```diff
@@ -771,6 +793,12 @@ def f(self, other, axis=default_axis, level=None, fill_value=None):
         if _should_reindex_frame_op(self, other, axis, default_axis, fill_value, level):
             return _frame_arith_method_with_reindex(self, other, op)
 
+        if not is_number(fill_value) and fill_value is not None:
```

Comment:
i dont think this condition is right.  what if the series we are working with is e.g. datetime64?